Le potato

fdanis-oss opened the pull request:
Add LePotato (libretech-dd) board image recipe

fdanis-oss committed:
LePotato: Initial minimal Debian image for AML-S905X-CC

The u-boot binary is provided in the image but needs some tweaks
before installing it, this is described in u-boot-amlogic documentation
(cf. u-boot/board/amlogic/p212/README.libretech-cc in u-boot sources)

Signed-off-by: Frédéric Danis <frederic.danis@collabora.com>

fdanis-oss committed:
LePotato: Add network support

Add NetworkManager

Signed-off-by: Frédéric Danis <frederic.danis@collabora.com>

fdanis-oss committed:
LePotato: Document the actions

Signed-off-by: Frédéric Danis <frederic.danis@collabora.com>

fdanis-oss committed:
LePotato: Add bmap block map file and compress final image

Signed-off-by: Frédéric Danis <frederic.danis@collabora.com>

fdanis-oss committed:
LePotato: Update README.md

Signed-off-by: Frédéric Danis <frederic.danis@collabora.com>

sjoerdsimons commented on lepotato/debimage-lepotato.yaml:
```diff
@@ -0,0 +1,87 @@
+  - action: run
+    description: Add localhost
+    chroot: true
+    command: echo "127.0.1.1\tdebian-{{ $suite }}-{{ $architecture }}" >> /etc/hosts
```
better to install `libnss-myhostname` then you don't have to mangle `/etc/hosts` as well as hostname

sjoerdsimons commented on lepotato/overlays/etc/hosts:
```diff
@@ -0,0 +1 @@
+127.0.0.1	localhost
```
should be dropped

sjoerdsimons commented on lepotato/overlays/etc/initramfs/post-update.d/99-uboot:
```diff
@@ -0,0 +1,7 @@
+
+echo "update-initramfs: Converting to u-boot format" >&2
+tempname="/boot/uInitrd-$1"
+mkimage -A arm64 -O linux -T ramdisk -C gzip -n uInitrd -d $2 $tempname > /dev/null
```
ugh you can use raw initramfs ifrom u-boot; no need to mangle them

sjoerdsimons commented on lepotato/overlays/etc/kernel/postinst.d/uboot:
```diff
@@ -0,0 +1,6 @@
+echo "kernel u-boot: install kernel link and dtb" >&2
+ln -sf $(basename $2) /boot/Image > /dev/null 2>&1
+cp -a /usr/lib/linux-image-$1 /boot/dtb > /dev/null 2>&1
+exit 0
```
same you can tell u-boot to use a directory of fdt files if the u-boot enable has been done correctly

sjoerdsimons commented on lepotato/overlays/boot/boot.cmd:
```diff
@@ -0,0 +1,70 @@
+# DO NOT EDIT THIS FILE
+#
+# Please edit /boot/armbianEnv.txt to set supported parameters
```
Does the lepotato not use the default distro boot? If it does better to install `u-boot-menu` instead of a big handcoded script

sjoerdsimons reviewed (dismissed):
Needs some polish (some of which might be becuse the le potato u-boot isn't great..). You also added a bunch of binary files without a clear licence; which is definately a no-no

andhe commented on lepotato/debimage-lepotato.yaml:
```diff
@@ -0,0 +1,68 @@
+{{- $architecture := or .architecture "arm64" -}}
+{{- $suite := or .suite "sid" -}}
```
Defaulting to `sid` seems like it'll likely sooner or later end up with the recipe breaking. If buster isn't new enough, then hopefully using `bullseye` would work and be more stable.

andhe commented on lepotato/debimage-lepotato.yaml:
```diff
@@ -0,0 +1,68 @@
+{{- $architecture := or .architecture "arm64" -}}
+{{- $suite := or .suite "sid" -}}
+{{ $type := or .type "min" }}
```
unused variable?!

andhe commented on lepotato/scripts/setup-user.sh:
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+USER=user
```
Not specific to this particular recipe, but a general comment/idea: Would be great if the username was a template/variable that could easily be overridden on debos command line. ie. something like using `USER=${1:-user}` in script and then `scripts/setup-user.sh {{ $adminusername }}` in recipe action (plus possibly `{{ $adminusername := or .adminusername "user" }}` at the top). (Also general shell scripting hygene: use `set -e ; set -u` ?!)

fdanis-oss committed:
LePotato: Move to bookworm and extract u-boot.bin

The u-boot.bin file is extracted from the target image to be used with
Amlogic binaries to create the bootloader image.

Signed-off-by: Frédéric Danis <frederic.danis@collabora.com>

obbardc commented on lepotato/README:
```diff
@@ -0,0 +1,6 @@
+The u-boot binary needs some tweaks before installing it, this is described in
```
@fdanis-oss The only blocker I see for this MR is to clarify what this means ? can we "tweak" the binary and install u-boot to the SD card inside the recipe?

fdanis-oss replied:
@obbardc Amlogic does not provide sources for the firmware and tools needed to create a
bootloader image so it is necessary to use some proprietary binaries. This is
described in [u-boot-amlogic documentation](https://github.com/u-boot/u-boot/blob/master/doc/board/amlogic/libretech-cc.rst).  
I didn't embed this in a debos action as binaries licenses have not clear licenses, as pointed by @sjoerdsimons in first comment.

I updated the README, move recipe to use Debian `bookworm` and add an action to extract the `u-boot.bin` file from the installed `u-boot-amlogic` package, as it is needed to create the bootloader image.

fdanis-oss commented:
@obbardc @sjoerdsimons I think all requested changes has been addressed. I there anything I should do to unblock it to be merged?
